mindocr/models/utils/attention_cells.py

```diff
@@ -23,29 +25,27 @@ def __init__(
         self.h = multi_attention_heads
         self.linears = nn.CellList([nn.Dense(dimensions, dimensions) for _ in range(4)])
         self.attention = None
-        self.dropout = nn.Dropout(keep_prob=1 - dropout)
+        if is_ms_version_2():
+            self.dropout = nn.Dropout(p=dropout)
+        else:
+            self.dropout = nn.Dropout(keep_prob=1 - dropout)
 
         self.matmul = ops.BatchMatMul()
 
     def dot_product_attention(
         self, query: Tensor, key: Tensor, value: Tensor, mask: Optional[Tensor] = None
     ) -> Tuple[Tensor, Tensor]:
 
-        d_k = value.shape[-1]
-        score = self.matmul(query, key.transpose(0, 1, 3, 2)) / ms.numpy.sqrt(
-            d_k
-        )  # (N, h, seq_len, seq_len)
-
-        # cast to fp32 to prevent overflow
-        score = ops.cast(score, ms.float32)
+        d_k = float(value.shape[-1])
```

Comment:
I think it should be `.float()` instead:
```shell
>>> test_var = ms.Tensor([1])
>>> type(test_var)
<class 'mindspore.common.tensor.Tensor'>
>>> type(test_var.float())
<class 'mindspore.common._stub_tensor.StubTensor'>
>>> type(float(test_var))
<class 'float'>
```
`float(Tensor)` gives constant as an output. This means that in the graph mode that constant will be calculated once only during graph construction and not changed during training.

Reply:
it is a integer (which corresponds to a constant value), it is not a tensor

Reply:
Then no problem!